phys2bids/interfaces/txt.py

```diff
@@ -10,10 +10,64 @@
 import numpy as np
 
 from phys2bids.physio_obj import BlueprintInput
+from operator import itemgetter
+from collections import Counter
 
 LGR = logging.getLogger(__name__)
 
 
+def check_multifreq(timeseries, freq, start=0, leftout=0):
+    """
+    Checks if there are channels with different frequency than the maximum one
+
+    Parameters
+    ----------
+    timeseries: list
+        list with channels only in np array format
+    freq : list
+        list with the maximun frequency
+    start : integer
+        first sample of the channel to be considered
+    leftout : integer
+        number of sample at the end of the channel that are not considered
+        This is done  so this process doesn't take forever
+    Returns
+    -------
+    mfreq: list
+        new list with the actual frequency of the channels
+    """
+    mfreq = []
+    # for each channel check frequency
+    max_equal = 1
+    for idx, chann in enumerate(timeseries):
+        eq_list = []
+        # cut the beggining of the channel
+        chann = chann[start:]
+        while len(chann) > max_equal:
+            eq_samples = 1  # start counter
+            for idx2, value in enumerate(chann[1:]):
+                # if value equal to previous value
+                if value == chann[idx2]:
+                    # count number of identic samples
+                    eq_samples += 1
+                else:
+                    # save this number when the next sample is not equal
+                    eq_list.append(eq_samples)
+                    # remove the samples that where equal
+                    chann = chann[idx2 + 1:]
+                    if max_equal < eq_samples:
+                        max_equal = eq_samples
+                    break
+        # count the number of ocurrences in eq_list
+        dict_fr = Counter(eq_list)
+        # get maximum
+        n_inter_samples = max(dict_fr.items(), key=itemgetter(1))[0]
+        # if there are interpolated samples, it means the frequency is lower
+        # decrease frequency by dividing for the number of equal samples continuos
```

Comment:
I do not understand this comment. Could you please elaborate a bit?

Reply:
Basically the frequency is divided by n_inter_samples which is the number of samples that are interpolated

Reply:
Cool! Thanks for the commit!